File: frontend/src/common/types/ai.ts
export interface Region {
  x: number;
  y: number;
  w: number;
  h: number;
}

export interface AnalyzeProcessedFace {
  region: Region;
  age: number;
  face_confidence: number;
  dominant_gender: string;
  gender: { [key: string]: number };
  dominant_emotion: string;
  emotion: { [key: string]: number };
  dominant_race: string;
  race: { [key: string]: number };
}

export interface FacialAreaWithEyes extends Region {
  left_eye?: [number, number];
  right_eye?: [number, number];
}

export interface ExtractFacesResult {
  facial_area: FacialAreaWithEyes;
  confidence: number;
  is_real?: boolean;
  antispoof_score?: number;
}

export interface FacialAreas {
  img1: Region;
  img2: Region;
}

export interface FaceVerificationResult {
  verified: boolean;
  distance: number;
  threshold: number;
  model: string;
  facial_areas: FacialAreas;
  time: number;
}

File: backend/src/schemas/ai.py
from pydantic import BaseModel


class Region(BaseModel):
    x: int
    y: int
    w: int
    h: int


class AnalyzeProcessedFace(BaseModel):
    region: Region
    age: float | None = None
    face_confidence: float | None = None
    dominant_gender: str | None = None
    gender: dict[str, float] | None = None
    dominant_emotion: str | None = None
    emotion: dict[str, float] | None = None
    dominant_race: str | None = None
    race: dict[str, float] | None = None


AnalyzeResponse = list[AnalyzeProcessedFace] | list


class AnalyzeResult(BaseModel):
    result: AnalyzeResponse


class FacialArea(BaseModel):
    x: int
    y: int
    w: int
    h: int
    left_eye: list[int]
    right_eye: list[int]


class FacialAreas(BaseModel):
    img1: Region
    img2: Region


class FaceDetectionResult(BaseModel):
    facial_area: FacialArea
    confidence: float
    is_real: bool | None = None
    antispoof_score: float | None = None


ExtractFacesResponse = list[FaceDetectionResult] | list


class ExtractFacesResult(BaseModel):
    result: ExtractFacesResponse


class VerifyResponse(BaseModel):
    verified: bool
    distance: float
    threshold: float
    model: str
    # distance_metric: str
    facial_areas: FacialAreas
    time: float
